conf_collectd_plugin.py

```diff
@@ -0,0 +1,375 @@
+    """
+    TODO:
+    -need to check if tcpconn plugin's dependencies are installed
+    -include RemotePort for outbounds connections
+        i.e. the servers we are connecting to
+    """
+    try:
+        out = open("10-tcpconns.conf", "w")
+    except:
+        sys.stderr.write("Unable to write tcpcons.conf file\n")
+        sys.exit(1)
+
+    out.write('LoadPlugin tcpconns')
+    out.write('<Plugin "tcpconns">\n')
+    out.write('  ListeningPorts false\n')
+    for port in open_ports:
+        out.write('  LocalPort "%d"\n' % port)
+    # no remote port yet
+    out.write('</Plugin>\n')
+    out.close()
+
+
+def include_apache_es_conf(dir):
+    """
+    Assume extendedstatus.conf is unique to wavefront and writing over it.
+    """
+    out = utils.write_file(dir+"/extendedstatus.conf")
+    if out is None:
+        utils.exit_with_message("Unexpected error!")
+
+    out.write(
+        '# ExtendedStatus controls whether Apache will generate '
+        '"full" status\n'
+        '# information (ExtendedStatus On) '
+        'or just basic information (ExtendedStatus\n'
+        '# Off) when the "server-status" handler is called. '
+        'The default is Off.\n'
+        'ExtendedStatus on')
+    out.close()
+
+
+# Apache Installation
+def apache_title():
+    sys.stdout.write(
+        " _____ __________  _____  _________   ___ ______________\n"
+        "   /  _  \\______   \/  _  \ \_   ___ \ /   |   \_   _____/\n"
+        "  /  /_\  \|     ___/  /_\  \/    \  \//    ~    \    __)_ \n"
+        " /    |    \    |  /    |    \     \___\    Y    /        \ \n"
+        " \____|__  /____|  \____|__  /\______  /\___|_  /_______  /\n"
+        "         \/                \/        \/       \/        \/ \n")
+
+
+def install_apache_plugin():
+    install = check_install_state("Apache")
+    print ""
+    if not install:
+        sys.stdout.write(
+            "This script has detected that you have apache installed and "
+            "running.")
+        res = utils.ask(
+            "Would you like to run the apache plugin installer to "
+            "enable collectd to collect data from apache?")
+    else:
+        print "You have previously installed this plugin."
+        res = utils.ask("Would you like to reinstall this plugin", no)
+
+    if not res:
+        return
+
+    apache_title()
+    print ""
+    sys.stdout.write(
+        "To enable collectd plugin with Apache, the following "
+        "steps need to be taken:\n"
+        "1. mod_status for apache needs to be enabled. (Default is enabled)\n"
+        "2. ExtendedStatus needs to be turned on.  (Default is off)\n"
+        "3. Enable the server-status handler for each virtual host.\n")
+
+    noop = raw_input("Press Enter to continue")
+    utils.print_step("Begin collectd Apache plugin installer")
+
+    """ check point
+    - check dependency
+    - change system file
+    - check mod status
+    - TODO: pull template
+    - prompt for user information
+    """
```

Comment:
yes

On Wed, Jul 6, 2016 at 1:18 PM, Jason Bau <notifications@github.com> wrote:

> In conf_collectd_plugin.py
> <https://github.com/wavefrontHQ/install/pull/22#discussion_r69803812>:
>
> > +        "To enable collectd plugin with Apache, the following "
> > +        "steps need to be taken:\n"
> > +        "1. mod_status for apache needs to be enabled. (Default is enabled)\n"
> > +        "2. ExtendedStatus needs to be turned on.  (Default is off)\n"
> > +        "3. Enable the server-status handler for each virtual host.\n")
> > +
> > +    noop = raw_input("Press Enter to continue")
> > +    utils.print_step("Begin collectd Apache plugin installer")
> > +
> > +    """ check point
> > +    - check dependency
> > +    - change system file
> > +    - check mod status
> > +    - TODO: pull template
> > +    - prompt for user information
> > +    """
>
> The above should be comment and not a string, right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/wavefrontHQ/install/pull/22/files/cd4dd95748662ae81a055276b6fdc466da95710a#r69803812>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ALZQ35wYaASgkjNF6ofp3cj5n0jUHrt2ks5qTA2cgaJpZM4JFrB2>
> .
>